src/custom-element.js

```diff
@@ -573,29 +573,32 @@ function createBaseCustomElementClass(win) {
 
     /**
      * Called to instruct the element to preconnect to hosts it uses during
-     * layout.
+     * layout. If attempting an early preconnect, it requests to preconnect
+     * to resources beforehand.
      * @param {boolean} onLayout Whether this was called after a layout.
      * @this {!Element}
      */
-    preconnect(onLayout) {
+    maybePreconnect(onLayout) {
       if (onLayout) {
         this.implementation_.preconnectCallback(onLayout);
       } else {
-        // If we do early preconnects we delay them a bit. This is kind of
-        // an unfortunate trade off, but it seems faster, because the DOM
-        // operations themselves are not free and might delay
-        Services.timerFor(toWin(this.ownerDocument.defaultView)).delay(() => {
-          const TAG = this.tagName;
-          if (!this.ownerDocument) {
-            dev().error(TAG, 'preconnect without ownerDocument');
-            return;
-          } else if (!this.ownerDocument.defaultView) {
-            dev().error(TAG, 'preconnect without defaultView');
-            return;
-          }
-          this.implementation_.preconnectCallback(onLayout);
-        }, 1);
+        this.getResources().requestPreconnect(this);
+      }
+    }
+
+    /**
+     * Preconnect to host.
+     */
+    preconnect() {
+      const TAG = this.tagName;
+      if (!this.ownerDocument) {
+        dev().error(TAG, 'preconnect without ownerDocument');
```

Comment:
This is a change in behavior. The element's ownerDocument could have been removed during the delay, not immediately.